Subject: [PATCH] Add Radio Tech support when search PLMN

Add radio technology interface when search PLMN. It works
in both cases when RIL reports radio tech or not.

diff --git a/telephony/java/com/android/internal/telephony/OperatorInfo.java b/telephony/java/com/android/internal/telephony/OperatorInfo.java
index a29d7c16199..7db660066eb 100644
--- a/telephony/java/com/android/internal/telephony/OperatorInfo.java
+++ b/telephony/java/com/android/internal/telephony/OperatorInfo.java
@@ -33,6 +33,7 @@
     private String mOperatorAlphaLong;
     private String mOperatorAlphaShort;
     private String mOperatorNumeric;
+    private String mRadioTech;
 
     private State mState = State.UNKNOWN;
 
@@ -57,6 +58,11 @@
         return mState;
     }
 
+    public String
+    getRadioTech() {
+        return mRadioTech;
+    }
+
     OperatorInfo(String operatorAlphaLong,
                 String operatorAlphaShort,
                 String operatorNumeric,
@@ -65,6 +71,14 @@
         mOperatorAlphaLong = operatorAlphaLong;
         mOperatorAlphaShort = operatorAlphaShort;
         mOperatorNumeric = operatorNumeric;
+        mRadioTech = "";
+        /* operatorNumeric format: PLMN+RAT or PLMN */
+        if (null != operatorNumeric) {
+            String values[] = operatorNumeric.split("\\+");
+            mOperatorNumeric = values[0];
+            if (values.length > 1)
+                mRadioTech = values[1];
+        }
 
         mState = state;
     }
@@ -108,6 +122,7 @@ public String toString() {
         return "OperatorInfo " + mOperatorAlphaLong
                 + "/" + mOperatorAlphaShort
                 + "/" + mOperatorNumeric
+                + "/" + mRadioTech
                 + "/" + mState;
     }
 
@@ -132,7 +147,7 @@ public int describeContents() {
     public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(mOperatorAlphaLong);
         dest.writeString(mOperatorAlphaShort);
-        dest.writeString(mOperatorNumeric);
+        dest.writeString(mOperatorNumeric + "+" + mRadioTech);
         dest.writeSerializable(mState);
     }
 
